thrust/thrust/detail/complex.inl

```diff
@@ -25,103 +25,86 @@ THRUST_NAMESPACE_BEGIN
 
 /* --- Equality Operators --- */
 
-template <typename T0, typename T1>
-__host__ __device__ typename detail::disable_if<detail::is_same<T0, T1>::value, bool>::type
-operator==(const complex<T0> &x, const complex<T1> &y)
+template <typename T0, typename T1, typename>
+__host__ __device__ bool operator==(const complex<T0> &x, const complex<T1> &y)
 {
   return x.real() == y.real() && x.imag() == y.imag();
 }
```

Comment:
I would prefer if we could just move those definitions up to the declaration. We do not really gain anything here

Reply:
Do you mean moving all the definitions in `complex.inl` to `complex.h` where they are declared ?

Reply:
yeah